gubernator.go

```diff
@@ -481,27 +488,59 @@ func (s *V1Instance) GetPeerRateLimits(ctx context.Context, r *GetPeerRateLimits
 
 	span.SetTag("numRequests", len(r.Requests))
 
-	var resp GetPeerRateLimitsResp
-
 	if len(r.Requests) > maxBatchSize {
 		err2 := fmt.Errorf("'PeerRequest.rate_limits' list too large; max size is '%d'", maxBatchSize)
 		ext.LogError(span, err2)
 		checkErrorCounter.WithLabelValues("Request too large").Add(1)
 		return nil, status.Error(codes.OutOfRange, err2.Error())
 	}
 
+	// Invoke each rate limit request in parallel.
+	var wg sync.WaitGroup
+	respChan := make(chan *RateLimitResp)
+
 	for _, req := range r.Requests {
-		rl, err := s.getRateLimit(ctx, req)
-		if err != nil {
-			// Return the error for this request
-			err2 := errors.Wrap(err, "Error in getRateLimit")
-			ext.LogError(span, err2)
-			rl = &RateLimitResp{Error: err2.Error()}
-			// checkErrorCounter is updated within getRateLimit().
-		}
-		resp.RateLimits = append(resp.RateLimits, rl)
+		wg.Add(1)
+		go func(req *RateLimitReq) {
+			defer wg.Done()
+			rl, err := s.getRateLimit(ctx, req)
+			if err != nil {
+				// Return the error for this request
+				err2 := errors.Wrap(err, "Error in getRateLimit")
+				ext.LogError(span, err2)
+				rl = &RateLimitResp{Error: err2.Error()}
+				// checkErrorCounter is updated within getRateLimit().
+			}
+
+			respChan <- rl
+		}(req)
 	}
-	return &resp, nil
+
+	// Capture each response.
+	resp := new(GetPeerRateLimitsResp)
+	var respWg sync.WaitGroup
+	respWg.Add(1)
+	go func() {
+		for rl := range respChan {
```

Comment:
I don't recall if the order of the rate limits matters for this call, I know it does for some of the other calls.

Reply:
You're right.  I fixed that.